### DISCUSSION TITLE
Can `firestore-backfire` export and import Firestore collections with lots of data?

### DISCUSSION THREAD

**Discussion by nonoumasy** [benyap/firestore-backfire#344] [Q&A] [answered]
I'm currently using this library: https://www.npmjs.com/package/node-firestore-import-export
It's no longer maintained and it has a flaw that it can't import large datasets.
Will this library do that?

All I want to do is to export my Firestore collection(which contains subcollections), do some data cleaning, and then import it back into Firestore. I'm looking for a simple solution.
Thanks in advance.

**Comment by benyap** [comment, marked as answer]
Hi @nonoumasy, yes this package does exactly that 😊

You can specify which collections you want to export, including nested subcollections (only if you want - it's configurable!).

You can run it on the command line or use it programatically.

The data is exported in [NDJSON](http://ndjson.org/) format. You can modify the data and import it right back into Firestore. I've used it to import and export many GBs of data and haven't come across any issues.

Usage instructions can be found on the [README](https://github.com/benyap/firestore-backfire#readme). If anything is unclear, please feel free to reach out!

**Reply by nonoumasy** [reply]
Thank you!
